File: src/analyzer/classify/classify-result.ts
import type { ReferenceFlagBits } from "../../ir/reference/reference-flags.js";

export type ClassifyResult =
  | { kind: "binding" }
  | { kind: "skip" }
  | {
      kind: "reference";
      flags: ReferenceFlagBits;
      init: boolean;
    };

File: crates/unsnarl-boundary-eslint-scope/src/classify/classify_result.rs
//! Result of classifying an identifier into a reference / binding /
//! skipped slot.
//!
//! Mirrors `ClassifyResult` in `classify/classify-result.ts`.

use unsnarl_ir::reference::reference_flags::ReferenceFlagBits;

pub(crate) enum ClassifyResult {
    Binding,
    Skip,
    Reference {
        flags: ReferenceFlagBits,
        init: bool,
    },
}
